src/ribasim_nl/ribasim_nl/cloud.py

```diff
@@ -0,0 +1,282 @@
+import logging
+import os
+import shutil
+from dataclasses import dataclass, field
+from pathlib import Path
+from typing import List, Tuple, Union
+from xml.etree import ElementTree
+
+import requests
+
+logger = logging.getLogger(__name__)
+
+RIBASIM_NL_CLOUD_PASS = os.getenv("RIBASIM_NL_CLOUD_PASS")
+RIBASIM_NL_DATA_DIR = os.getenv("RIBASIM_NL_DATA_DIR")
+RIBASIM_NL_CLOUD_USER = "nhi_api"
+WEBDAV_URL = "https://deltares.thegood.cloud/remote.php/dav"
+BASE_URL = rf"{WEBDAV_URL}/files/{RIBASIM_NL_CLOUD_USER}/Ribasim modeldata"
+
+WATER_AUTHORITIES = [
+    "AaenMaas",
+    "AmstelGooienVecht",
+    "BrabantseDelta",
+    "DeDommel",
+    "Delfland",
+    "DrentsOverijsselseDelta",
+    "HollandseDelta",
+    "HollandsNoorderkwartier",
+    "HunzeenAas",
+    "Limburg",
+    "Noorderzijlvest",
+    "RijnenIJssel",
+    "Rijnland",
+    "Rivierenland",
+    "Rijkswaterstaat",
+    "Scheldestromen",
+    "SchielandendeKrimpenerwaard",
+    "StichtseRijnlanden",
+    "ValleienVeluwe",
+    "Vechtstromen",
+    "WetterskipFryslan",
+    "Zuiderzeeland",
+]  # noqa
+
+HIDDEN_DIRS = ["D-HYDRO modeldata"]  # somehow this dir-name still exists :-(
+
+
+def is_dir(item):
+    return Path(item).suffix == ""
+
+
+@dataclass
+class Cloud:
```

Comment:
I think I'd prefer `CloudStorage` here, since `Cloud` could mean pretty much anything. And it would be good to have a docstring.